swag/tools.py

```diff
@@ -41,59 +62,170 @@ def validate_types(self) -> Self:
 
         return self
 
+
+class GetDirections(BaseModel):
+    """Get directions between two locations."""
+
+    origin: str = Field(description="The starting location")
+    destination: str = Field(description="The destination location")
+    mode: str = Field(
+        description="The mode of transportation (driving, walking, bicycling, transit)",
+        default="driving",
+    )
+
+
+class GetDistanceMatrix(BaseModel):
+    """Get a matrix of distances between multiple origins and destinations."""
+
+    origins: list[tuple[float, float]] = Field(
+        description="List of origin locations as tuples of (latitude, longitude)"
+    )
+    destinations: list[tuple[float, float]] = Field(
+        description="List of destination locations as tuples of (latitude, longitude)"
+    )
+    mode: str = Field(
+        description="The mode of transportation (driving, walking, bicycling, transit)",
```

Comment:
I think some potential options should be added here to quickly fail if the model get it's wrong.

Reply:
Right, updated